backend/app/store/search/search_test.go

```diff
@@ -0,0 +1,333 @@
+package search
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"os"
+	"testing"
+	"time"
+
+	log "github.com/go-pkgz/lgr"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/umputun/remark42/backend/app/store"
+	"github.com/umputun/remark42/backend/app/store/engine"
+	bolt "go.etcd.io/bbolt"
+)
+
+func createTestService(t *testing.T, sites []string) (searcher Service, teardown func()) {
+	tmp := os.TempDir()
+	idxPath := tmp + "/search-remark42"
+
+	_ = os.RemoveAll(idxPath)
+
+	searcher, err := NewSearcher(SearcherParams{
+		Type:      "bleve",
+		IndexPath: idxPath,
+		Analyzer:  "standard",
+		Sites:     sites,
+	})
+
+	require.NoError(t, err)
+
+	teardown = func() {
+
+		err := searcher.Close()
+		require.NoError(t, err)
+		_ = os.RemoveAll(idxPath)
+	}
+	return searcher, teardown
+}
+
+func TestSearch_SiteMux(t *testing.T) {
+	searcher, teardown := createTestService(t, []string{"test-site", "test-site2", "test-site3"})
+	defer teardown()
+	err := searcher.Init(context.Background(), nil)
+	assert.NoError(t, err)
+
+	searcher.IndexDocument("123456", &store.Comment{
+		ID:        "123456",
+		Locator:   store.Locator{SiteID: "test-site", URL: "http://example.com/post1"},
+		Text:      "text 123",
+		User:      store.User{ID: "u1", Name: "user1"},
+		Timestamp: time.Date(2017, 12, 20, 15, 18, 24, 0, time.Local),
+	})
+
+	searcher.IndexDocument("123456", &store.Comment{
+		ID:        "123456",
+		Locator:   store.Locator{SiteID: "test-site2", URL: "http://example.com/post1"},
+		Text:      "text 345",
+		User:      store.User{ID: "u1", Name: "user1"},
+		Timestamp: time.Date(2017, 12, 20, 15, 20, 24, 0, time.Local),
+	})
+
+	searcher.IndexDocument("123457", &store.Comment{
+		ID:        "123457",
+		Locator:   store.Locator{SiteID: "test-site2", URL: "http://example.com/post1"},
+		Text:      "foobar 345",
+		User:      store.User{ID: "u1", Name: "user1"},
+		Timestamp: time.Date(2017, 12, 20, 15, 20, 28, 0, time.Local),
+	})
+
+	_ = searcher.Flush("test-site")
```

Comment:
It doesn't seem right to not check for errors in such a simple to check place in test.